### DISCUSSION TITLE
Assets loading in a loop when deploying behind subpath (nginx)

### DISCUSSION THREAD

**Discussion by junpls** [storybookjs/storybook#16065] [Help] [answered]
Hey all!

I want to deploy Storybook (for Angular) behind the subpath `/storybook/`.
This is my nginx configuration entry:

```
location /storybook/ {
  alias /var/www/dist/storybook;
  index  index.html index.htm;
  try_files $uri$args $uri /index.html =404;
  break;
}

```

What now happens is, that the overall Storybook app is loading, but the stories don't:
![grafik](https://user-images.githubusercontent.com/9803353/133411556-6ecfde52-eb49-42ce-9826-fac64f18bbd1.png)

From what I can see in the network log, the assets just keep loading in a loop:
![grafik](https://user-images.githubusercontent.com/9803353/133411871-4d35e9ee-7384-483c-a471-47e04d7a184f.png)
The data returned in each request looks like the correct html/js, so not something that nginx generated.

Interestingly, the URL for the iframe gains more and more query parameters each time:
`https://localhost/storybook/iframe.html?id=*&viewMode=story&id[0]=*&id[1]=*&id[2]=*&id[3]=*&id[4]=*&id[5]=*&id[6]=*&id[7]=*&id[8]=*&id[9]=*&id[10]=*&id[11]=*&id[12]=*&id[13]=*&id[14]=*&id[15]=*&id[16]=*&viewMode[0]=story&viewMode[1]=story&viewMode[2]=story&viewMode[3]=story&viewMode[4]=story&viewMode[5]=story&viewMode[6]=story&viewMode[7]=story&viewMode[8]=story&viewMode[9]=story&viewMode[10]=story&viewMode[11]=story&viewMode[12]=story&viewMode[13]=story&viewMode[14]=story&viewMode[15]=story&viewMode[16]=story`

What I already tried is basically everying menioned in this thread [https://github.com/storybookjs/storybook/issues/1291](https://github.com/storybookjs/storybook/issues/1291), but I reset it all to the defaults now, as it had no effect.

```
Environment Info:

  Binaries:
    Node: 14.17.0 - /usr/local/opt/node@14/bin/node
    npm: 6.14.13 - /usr/local/opt/node@14/bin/npm
  npmPackages:
    @storybook/addon-actions: ^6.3.8 => 6.3.8 
    @storybook/addon-essentials: ^6.3.8 => 6.3.8 
    @storybook/addon-links: ^6.3.8 => 6.3.8 
    @storybook/angular: ^6.3.8 => 6.3.8 
```

What I should maybe also mention is, that it works perfectly fine when deploying without a subpath.

**Comment by wprzyb** [comment]
Also would love to get some advice on that matter, same problem.

**Comment by junpls** [comment, marked as answer]
@wprzyb 

Alright, I got it to work on storybook 6.4.18.

There were two problems in my setup:

1. I built storybook using `--preview-url=/storybook/iframe.html` as described [here](https://github.com/storybookjs/storybook/issues/1291#issuecomment-795251283). This resulted in no **_iframe.html_** being generated. The modifications to **_.storybook/main.js_** described there seem to have worked though. `--preview-url=/storybook/iframe.html` was simply not needed.
2. Missing trailing `/` in the alias directive: `alias /var/www/dist/storybook/;`

Both of these factors resulted in nginx defaulting to serve `index.html` instead of `iframe.html`.
